tests/system/src/test/java/test/com/sun/javafx/text/TextLayoutTest.java

```diff
@@ -0,0 +1,604 @@
+
+        /**
+         * Checks that single white spaces are ignored for alignment
+         * purposes when wrapping is enabled in simple text. This soft
+         * wraps after "jumps".
+         */
+        SOFT_WRAP(
+            "The quick brown fox jumps over the lazy dog",
+            145.0f, List.of(142.72852f, 91.38867f)
+        ),
+
+        /**
+         * Checks that leading white spaces are NOT stripped even when
+         * some lines are soft wrapped. This soft wraps after "fox"
+         * and has 4 leading spaces on the 1st line where no soft
+         * wrap occurs, and so they should be kept intact.
+         */
+        SOFT_WRAP_WITH_LEADING_SPACE(
+            "    The quick brown fox jumps over the lazy dog",
+            145.0f, List.of(107.384762f + 4 * ARIAL_SPACE_ADVANCE, 126.73242f)
+        ),
+
+        /**
+         * Checks that trailing white spaces are NOT stripped even when
+         * some lines are soft wrapped. This soft wraps after "jumps"
+         * and has 4 trailing spaces on the 2nd line where no soft
+         * wrap occurs, and so they should be kept intact.
+         */
+        SOFT_WRAP_WITH_TRAILING_SPACE(
+            "The quick brown fox jumps over the lazy dog    ",
+            145.0f, List.of(142.72852f, 91.38867f + 4 * ARIAL_SPACE_ADVANCE)
+        ),
+
+        /**
+         * Checks that multiple white spaces are ignored for alignment
+         * purposes when wrapping is enabled in simple text. This soft
+         * wraps after "jumps".
+         */
+        SOFT_WRAP_WITH_EXTRA_SPACE(
+            "The quick brown fox jumps           over the lazy dog",
+            145.0f, List.of(142.72852f, 91.38867f)
+        ),
+
+        /**
+         * Checks that single white spaces are ignored for alignment
+         * purposes when wrapping is enabled in complex text. This soft
+         * wraps after "jumps".
+         */
+        SOFT_WRAP_WITH_COMPLEX_TEXT_ON_WINDOWS(
+            TextLayoutTest::assumeWindows,
+            "The quick brown लोमड़ी jumps over the lazy कुत्ता",
+            160.0f, List.of(158.1914f, 93.134766f)
+        ),
+
+        /**
+         * Checks that multiple white spaces are ignored for alignment
+         * purposes when wrapping is enabled in complex text. This soft
+         * wraps after "jumps".
+         */
+        SOFT_WRAP_WITH_COMPLEX_TEXT_AND_EXTRA_SPACE_ON_WINDOWS(
+            TextLayoutTest::assumeWindows,
+            "The quick brown लोमड़ी jumps           over the lazy कुत्ता",
+            160.0f, List.of(158.1914f, 93.134766f)
+        ),
+
+        /**
+         * Checks that single white spaces are ignored for alignment
+         * purposes when wrapping is enabled in complex text. This soft
+         * wraps after "jumps".
+         */
+        SOFT_WRAP_WITH_COMPLEX_TEXT_ON_MAC(
+            TextLayoutTest::assumeMac,
+            "The quick brown लोमड़ी jumps over the lazy कुत्ता",
+            160.0f, List.of(155.3047f, 91.04719f)
```

Comment:
Exact comparisons on float numbers should not be a problem, as long as you don't assume that with a different order of operations you will get the same result.  For example, `y + x - y` may not exactly equal `x + y - y` or `x`.  Since we don't have different code paths for different platforms, and since all code paths are deterministic(*), it should be fine to do an exact comparison. 

If this ever does fail, I think then is a good time to do a more thorough investigation. So far these tests have run dozens of time on my machine, and on several of the machines of people reviewing without problems.

(*) In tests, the most often encountered source of non-determinism in Java is when a `Set` or `Map` is involved which contains objects with the default hash code which can change each time a new JVM is created; usually I change such collections to linked versions, or do an explicit sort somewhere to ensure it is deterministic again.

Reply:
> Question: these numbers are compared exactly (Float::compare), can we expect them to be exactly the same on all possible platforms/versions?

Sorry, I stopped doing that, I thought there was an automated script for this?